src/Services/Database/TransactionService.php

```diff
@@ -38,13 +37,13 @@ public function get(string $key, string $column = 'id'): Model
     /**
      * Create a new transaction.
      */
-    public function store(array $data, ?Model $signingWallet = null): Model
+    public function store(array $data, string|Model $signingWallet = null): Model
     {
         if ($transaction = Transaction::firstWhere(['idempotency_key' => $data['idempotency_key']])) {
             return $transaction;
         }
 
-        $data['wallet_public_key'] = $signingWallet->public_key ?? Account::daemon()->public_key;
+        $data['wallet_public_key'] = is_string($signingWallet) ? $signingWallet : $signingWallet->public_key;
```

Comment:
We are not handling the case where 2nd parameter is just null, it can lead to crash

Reply:
thanks